### PR TITLE
WebDAV: fix bugs when start folder starts with "/"

### PR DESCRIPTION
**Pull request by weijiuqiao**
This PR fixes bugs in `webdavapi`:
- error when url ends with `/` and start folder starts with `/`
- error checking current folder when url encode has different letter case than server

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9688)

### CODE REVIEW

File: frontend/apps/cloudstorage/webdavapi.lua [hunk 1]
```diff
@@ -11,6 +11,13 @@ local logger = require("logger")
 local WebDavApi = {
 }
 
+function WebDavApi:jointPath( address, path )
```
Comment (Frenzie) [inline frontend/apps/cloudstorage/webdavapi.lua:14]: joined?

Comment (poire-z) [reply frontend/apps/cloudstorage/webdavapi.lua]: Or, as it is a method and better be a verb: :joinPath() or :getJoinedPath() :)

Comment (weijiuqiao) [reply frontend/apps/cloudstorage/webdavapi.lua]: Done again

**Review by Frenzie — APPROVED** [review]
lgtm

**Comment by weijiuqiao** [comment]
Is this mergable? Cause I want to put out Cloud sync for `statistics` and `vocabbuilder` plugins, which is on top of this PR, and get some comments and/or guidance. :)

**Pull request merged**